src/plugins/formulas/formulas.js

```diff
@@ -197,64 +265,198 @@ export class Formulas extends BasePlugin {
-    if (!this.enabled) {
+  /**
+   * Add a sheet to the shared HyperFormula instance.
+   *
+   * @param {string} [sheetName] The new sheet name. If not provided, will be auto-generated by HyperFormula.
+   * @param {Array} sheetData Data passed to the shared HyperFormula instance. Has to be declared as an array of
+   * arrays - array of objects is not supported in this scenario.
+   * @returns {boolean} `false` if the data format is unusable, `true` otherwise.
+   */
+  addSheet(sheetName, sheetData) {
+    if (!isAoA(sheetData)) {
+      error('The provided data should be an array of arrays.');
+
+      return false;
+    }
+
+    if (this.engine.doesSheetExist(sheetName)) {
+      error('Sheet with the provided name already exists.');
+
+      return false;
+    }
+
+    const actualSheetName = this.engine.addSheet(sheetName ?? void 0);
+
+    if (this.engine.isItPossibleToReplaceSheetContent(actualSheetName, sheetData)) {
+      this.engine.setSheetContent(actualSheetName, sheetData);
+    }
+
+    return true;
+  }
+
+  /**
+   * Switch the sheet used as data in the Handsontable instance (it loads the data from the shared HyperFormula
+   * instance).
+   *
+   * @param {string} sheetName Sheet name used in the shared HyperFormula instance.
+   */
+  switchSheet(sheetName) {
+    this.sheetName = sheetName;
+
+    this.hot.loadData(this.engine.getSheetSerialized(this.sheetId), this.#pluginSourceString);
+  }
+
+  /**
+   * `beforeLoadData` hook callback.
+   *
+   * @param {Array} sourceData Array of arrays or array of objects containing data.
+   * @param {boolean} initialLoad Flag that determines whether the data has been loaded during the initialization.
+   * @param {string} source Source of the call.
+   * @private
+   */
+  onBeforeLoadData(sourceData, initialLoad, source) {
+    if (source === this.#pluginSourceString) {
       return;
     }
 
-    this.hyperformula.setSheetContent(this.sheetName, this.hot.getSourceDataArray());
+    // This flag needs to be defined, because not passing data to HOT results in HOT auto-generating a `null`-filled
+    // initial dataset.
+    this.#hotWasInitializedWithEmptyData = isUndefined(this.hot.getSettings().data);
   }
 
+  /**
+   * `afterLoadData` hook callback.
+   *
+   * @param {Array} sourceData Array of arrays or array of objects containing data.
+   * @param {boolean} initialLoad Flag that determines whether the data has been loaded during the initialization.
+   * @param {string} source Source of the call.
+   * @private
+   */
+  onAfterLoadData(sourceData, initialLoad, source) {
+    if (source === this.#pluginSourceString) {
+      return;
+    }
+
+    const sheetName = this.#settings.sheetName;
+
+    if (
+      isUndefined(sheetName) ||
+      (isDefined(sheetName) && !this.engine.doesSheetExist(sheetName))
+    ) {
+      this.sheetName = this.engine.addSheet(sheetName);
+
+    } else if (isDefined(sheetName)) {
+      this.sheetName = sheetName;
+    }
```

Comment:
Or would reversing the `if` make it more clear that `else` does not have any condition and we will always do something and have the existing sheet?
```suggestion
    if (isDefined(sheetName) && this.engine.doesSheetExist(sheetName)) {
      this.sheetName = sheetName;

    } else {
      this.sheetName = this.engine.addSheet(sheetName);
    }
```